packages/manager/core/application/src/hooks/useBreadcrumb.tsx

```diff
@@ -0,0 +1,109 @@
+/* eslint-disable @typescript-eslint/no-use-before-define */
+/* eslint-disable import/no-extraneous-dependencies */
+/* eslint-disable @typescript-eslint/naming-convention */
+import React, { useEffect, useState } from 'react';
+import { Params, useMatches } from 'react-router-dom';
+import { OsdsBreadcrumb } from '@ovhcloud/ods-components/breadcrumb/react';
+import { useShell } from '..';
+
+export type useBreadcrumbHandleParams = {
+  data: unknown;
+  params: Params<string>;
+};
+
+export type useBreadcrumbItem = {
+  label: string | undefined;
+  href: string;
+};
+
+export type useBreadcrumbHandle = (
+  params: useBreadcrumbHandleParams,
+) => Promise<string>;
+
+type MatchHandle = {
+  breadcrumb?: useBreadcrumbHandle;
+};
+
+export interface useBreadcrumbProps {
+  rootLabel?: string;
+}
+
+export const useBreadcrumb = ({ rootLabel }: useBreadcrumbProps) => {
+  const matches = useMatches();
+  const shell = useShell();
+  const [root, setRoot] = useState<useBreadcrumbItem[]>([]);
+  const [matchCrumbs, setMatchCrumbs] = useState<useBreadcrumbItem[]>([]);
+
+  useEffect(() => {
+    const fetchRoot = async () => {
+      try {
+        const response = await shell.navigation.getURL('hub', '#/', {});
+        console.log(response);
+        if (typeof response === 'string') {
+          setRoot([
+            {
+              label: rootLabel,
+              href: response,
+            },
+          ]);
+        } else {
+          console.error('Invalid response type:', typeof response);
+        }
+      } catch (error) {
+        console.error('Error fetching root URL:', error);
+      }
+    };
+
+    fetchRoot();
+  }, [rootLabel, shell.navigation]);
+
+  useEffect(() => {
+    const fetchMatchCrumbs = async () => {
+      try {
+        const items = await Promise.all(
+          matches.map(async (match) => {
+            const { handle, data, params } = match;
+
+            if (isMatchHandle(handle)) {
+              const breadcrumb = await Promise.resolve(handle.breadcrumb);
```

Comment:
```suggestion
              const { breadcrumb } = handle;
```

Reply:
I don't understand where the "basic crumb" is called on this line

Reply:
if handle.breadcrumb is potentially a promise, we should just `await handle.breadcrumb`
`await Promise.resolve(x)` is equivalent to `x`